umd-annotation/umd-annotation-tool
# Watchtower Manual Updates

resolves #36 

Realized the port doesn't need to be exposed and can use an admin guarded Girder rest endpoint to initiate the watcher update.

Testing procedure:

1. I had not updated or rebuilt my containers in a while so I knew that UI (Vue App) was out of date when compared to the latest release on github.
2. Launched the server normally and took a screenshot of the githash: 
![image](https://user-images.githubusercontent.com/61746913/213705138-649bd193-121a-4046-9a66-09f1b9962bae.png)
3.  Navigated to the Girder Swagger rest endpoints, logged in as and admin and used the POST UMD_dataset/update_containers endpoint to trigger an update.
4. Watched the Watchdtower container logs to see that it was trying to pull the latest docker containers from the ghcr.io (github container registry).
5. After it was complete it restarter the containers and I could check for the latest git hash: 
![image](https://user-images.githubusercontent.com/61746913/213705465-ad75c6c8-4f8b-4ad1-bbe7-60328016f047.png)
This hash matches the latest github commit for the system.

So it looks like it works and utilizing another environment flag: `- WATCHTOWER_HTTP_API_PERIODIC_POLLS=true`
keeps the periodic checking for Sun early morning at 2 A.M.
